### ISSUE TITLE
Plans Generation

### ISSUE THREAD

**Issue by pandanwaffles** [matsim-org/matsim-code-examples#669]
Hi, I have been trying to simulate the traffic through MATSim. Currently I’ve been able to get the network.xml of an area of interest which I'd like to work on, however I am unable to figure out how to generate a plans/population for the network. I have tried out the codes found here [https://github.com/matsim-org/matsim-code-examples/tree/13.x/src/main/java/org/matsim/codeexamples/population/demandGeneration](https://github.com/matsim-org/matsim-code-examples/tree/13.x/src/main/java/org/matsim/codeexamples/population/demandGeneration) but the generated plans do not match the nodes and links in my XML file. I would like to enquire the process to generate a plans for the vehicles/agents on the existing network which I have.

**Comment by Janekdererste** [comment]
It is not necessary to reference the network in your plans file. If you know the coordinates of your activities it is sufficient to give each activity a coordinate. Also one need to specify legs inbetween activities. These only require a mode.

Once MATSim is started, it will map the activities to the closest link of the network. Also, the legs receive routes which then reference the network on startup. 

The following example gives an overview on what is necessary for a minimal plan. The snippet is taken from [here](https://github.com/matsim-org/matsim-code-examples/blob/13.x/src/main/java/org/matsim/codeexamples/population/demandGenerationFromShapefile/CreateDemand.java)

```
// this is how coordinates are initialized. Usually the actual data is retreived from somewhere else, e.g. a survey
Coord home = new Coord(51.0, 32.0);
Coord work = new Coord(52.0,32.0);

// create a plan for home and work. Note, that activity -> leg -> activity -> leg -> activity have to be inserted in the right
// order.
Plan plan = population.getFactory().createPlan();

Activity homeActivityInTheMorning = population.getFactory().createActivityFromCoord("home", home);
homeActivityInTheMorning.setEndTime(HOME_END_TIME);
plan.addActivity(homeActivityInTheMorning);

// this creates a generic leg inbetween two activities with the mode you specify here
Leg toWork = population.getFactory().createLeg("car");
plan.addLeg(toWork);

Activity workActivity = population.getFactory().createActivityFromCoord("work", work);
workActivity.setEndTime(WORK_END_TIME);
plan.addActivity(workActivity);

Leg toHome = population.getFactory().createLeg(mode);
plan.addLeg(toHome);

Activity homeActivityInTheEvening = population.getFactory().createActivityFromCoord("home", home);
plan.addActivity(homeActivityInTheEvening);

**Comment by pandanwaffles** [comment]
Hi,
Thank you for the reply.

According to the code from [here](https://github.com/matsim-org/matsim-code-examples/blob/13.x/src/main/java/org/matsim/codeexamples/population/demandGenerationFromShapefile/CreateDemand.java), 

		this.interRegionCommuterStatistic = sampleFolder.resolve("commuters-inter-regional.csv");
		this.innerRegionCommuterStatistic = sampleFolder.resolve("commuters-inner-regional.csv");

		// read in the shape file and store the geometries according to their region identifier stored as 'RS' in the
		// shape file
		regions = ShapeFileReader.getAllFeatures(sampleFolder.resolve("thueringen-kreise.shp").toString()).stream()
				.collect(Collectors.toMap(feature -> (String) feature.getAttribute("RS"), feature -> (Geometry) feature.getDefaultGeometry()));

		// Read in landcover data to make people stay in populated areas
		// we are using a weighted distribution by area-size, so that small areas receive less inhabitants than more
		// populated ones.
		List<Pair<Geometry, Double>> weightedGeometries = new ArrayList<>();
		for (SimpleFeature feature : ShapeFileReader.getAllFeatures(sampleFolder.resolve("landcover.shp").toString())) {
			Geometry geometry = (Geometry) feature.getDefaultGeometry();
			weightedGeometries.add(new Pair<>(geometry, geometry.getArea()));
		}
		landcover = new EnumeratedDistribution<>(weightedGeometries);

		this.population = PopulationUtils.createPopulation(ConfigUtils.createConfig());
	}

	Population getPopulation() {
		return this.population;
	}`

Do I have to replace the shapefile `"thueringen-kreise.shp"` with a shapefile of the area I'm working on?

And may I ask what is does this part below mean and do you have a copy of the 2 csv for my reference?
```
	this.interRegionCommuterStatistic = sampleFolder.resolve("commuters-inter-regional.csv");
	this.innerRegionCommuterStatistic = sampleFolder.resolve("commuters-inner-regional.csv");
```

Thank you so much

**Comment by Janekdererste** [comment]
So, for the csv files, the class has a lengthy comment at the top which describes, where the data comes from. 
```
	this.interRegionCommuterStatistic = sampleFolder.resolve("commuters-inter-regional.csv");
	this.innerRegionCommuterStatistic = sampleFolder.resolve("commuters-inner-regional.csv");
```
This code snippet simple resolves the input folder and the csv-filenames to a filepath. Those are used later. 

The entire class is not necessarily meant to be taken as is. It just combines a lot of thing one typically has to solve when generating a population from a commuter statistics. Depending on your data source for agent activities, your script to convert those into matsim plans will probably look a lot different.

**Comment by pandanwaffles** [comment]
Thank you so much for the reply and help! 

Just to confirm, I would have to have a commuter statistics as data source in order to generate the plans? 
Would traffic flow of heavy vehicles and light vehicles work?

Thank you once again!

**Comment by Janekdererste** [comment]
Yes, you need some form of data. Commuter statistics, or traffic flows are a good starting point. Trajectories from cell phone data or travel diaries, are would be even closer to the mental model of travelling in MATSim. 

For the commuter statistics and the traffic flows: Assuming that both contain flows from origin to destination areas, one needs to generate actual coordinates for the activities from those as it is done in the script above.

**Comment by pandanwaffles** [comment]
Hi,
I came across this source.
https://www.simunto.com/matsim/tutorials/eifer2019/slides_day2.pdf
![image](https://user-images.githubusercontent.com/90554682/150989396-b47ab3d7-ac43-4a08-b3e5-f575f93bb1bc.png)

Hence, I would like to check if MATSim API only works for test scenarios? 
Am I able to use the MATSim API to create my own output simulation?

Thank you!

**Comment by Lucieeef** [comment]
Hi, did you eventually find a way to generate the plans file?
I have the same problem.

**Comment by shairys88-Malmo** [comment]
Hi @Janekdererste 
I have kind of same question but with different scenario. 

### **Problem**: I need to generate plans based on CSV/excel file.


_Here is the detail:_
I have DRT database for Sweden case with approx 50,000 entries. Headers in the database are shown below:

![Image](https://github.com/user-attachments/assets/5a445b09-7383-4953-b40d-2e0a9daa93b9)

I presume, for now during the plans generation, I only need the highlighted one. (Status/passengerstatus is only to check whether the journey was completed or cancelled)

I am using this code [here](https://github.com/MariaLucia813/matsim-code-examples/blob/11.x/src/main/java/org/matsim/codeexamples/population/CreatingPopulationCSV) 

**Queries:**
1. I don't have the shape file, do I need it?
2. Can I convert my CSV/excel to plans using this, am I using the right code? 
3. If so, what major changes do I need to make to convert my CSV/excel to plans.
4. If not, direct me to the right repository, please.

**Annex:**
_For now I am writing the plans by hand (but daily request were few). Now I have to go to 150-200 request per day, then I need to convert the CSV/excel to plans._

_Here is what I used before, by writing one by one._ 

```
        <person id="105">
		<plan selected="yes">
			<activity type=dfs"home" x="405581.40" y="6406856.54" end_time="07:49:00"/>
			<leg mode="drt"/>
			<activity type="work" x="405768.31" y="6406817.19"/>
		</plan>
	</person>

	<person id="106">
		<plan selected="yes">
			<activity type="home" x="405581.40" y="6406856.54" end_time="09:27:00"/>
			<leg mode="drt"/>
			<activity type="work" x="407154.97" y="6406016.56" end_time="19:02:00"/>
			<leg mode="drt"/>
			<activity type="work" x="405720.66" y="6409815.11"/>
		</plan>
	</person>

	<person id="107">
		<plan selected="yes">
			<activity type="home" x="405790.11" y="6407004.77" end_time="08:35:00"/>
			<leg mode="drt"/>
			<activity type="work" x="405720.66" y="6409815.11" end_time="08:48:00"/>
			<leg mode="drt"/>
			<activity type="work" x="407154.97" y="6406016.56" end_time="09:29:00"/>
			<leg mode="drt"/>
			<activity type="work" x="405714.06" y="6406613.17" end_time="10:18:00"/>
			<leg mode="drt"/>
			<activity type="work" x="404777.43" y="6404136.93"/>
		</plan>
	</person>
```

Thanks in advance. 
Regards
~Malik

**Comment by Janekdererste** [comment]
> I don't have the shape file, do I need it?

I don't think so.

> Can I convert my CSV/excel to plans using this, am I using the right code?

I don't think this code is tailored to generating a population from scratch. [This](https://github.com/matsim-org/matsim-code-examples/blob/2024.x/src/main/java/org/matsim/codeexamples/population/demandGenerationFromShapefile/CreateDemand.java) example also uses data from a CSV-file and converts it into plans. So, maybe this is a better fit. That said, your example uses MATSim v11.x (6 years old). You could fork the current version of the example-project instead, which uses v2024.0. 

The question here is, I think, whether your data is the correct basis for generating the traffic demand. You just have pickup/dropoff data. If you are only interested in simulating DRT and nothing else, you can certainly create a population with activities at those locations and then simulate how your DRT-Service serves these persons. If you want to simulate regular traffic, for example other cars on the road, you probably need other data sources. https://github.com/matsim-org/matsim-code-examples/issues/1173 might be related to this.
